src/lib/PDF.cpp

```diff
@@ -175,7 +221,50 @@ pair<double, double> PDF::uncMin(double dchi) {
   }
 
   // Loop over each point of the PDF to find the maximum error
-  for (size_t k = ib; k < xaxis.size(); k++) {
+  for (size_t k = ib; k < xaxis.size() - 1; k++) {
+    // When the considered chi2 is above the chi2 limit and the following one
+    // below
+    if (chi2[k] <= chiLim && chi2[k + 1] > chiLim) {
+      // Linear interpolation
+      double slope = (chiLim - chi2[k]) / (chi2[k + 1] - chi2[k]);
+      // store the minimum error
+      result.second = xaxis[k] + slope * (xaxis[k + 1] - xaxis[k]);
+      break;
+    }
+  }
+
+  return result;
+}
+
+pair<double, double> PDF::confidence_interval(float dchi) {
+  // Initialize the uncertainties with the range considered
+  pair<double, double> result = make_pair(xaxis[0], xaxis[vsize - 1]);
+
+  // get the chi2 minimum with quadratic improvement
+  auto extremum = improve_extremum(true);
+  double zmin = extremum.first;
+  double chi2min = extremum.second;
+
+  size_t min_ib = index(zmin);
+  double chiLim = chi2min + dchi;
+
+  // Now we search for the position where the chi2 curve goes above chiLim,
+  // right and left from index min_ib, which remains reasonable compared to zmin
+  // position Loop over each point of the PDF to find the minimum error
+  for (int k = 0; k < min_ib; k++) {
+    // When the considered chi2 is below the chi2 limit and the following one
+    // above
+    if (chi2[k] >= chiLim && chi2[k + 1] < chiLim) {
+      // Linear interpolation
+      double slope = (chiLim - chi2[k]) / (chi2[k + 1] - chi2[k]);
+      // store the minimum error
+      result.first = xaxis[k] + slope * (xaxis[k + 1] - xaxis[k]);
+      break;
+    }
+  }
+
+  // Loop over each point of the PDF to find the maximum error
```

Comment:
This comment now seems like a duplicate given the additional loop?

Reply:
not sure to follow : the first loop searches left of chimin, while the second one searches right of chimin.